api/lib/opentelemetry/trace/link.rb

```diff
@@ -1,43 +0,0 @@
-# frozen_string_literal: true
-
-# Copyright 2019 OpenTelemetry Authors
-#
-# SPDX-License-Identifier: Apache-2.0
-
-module OpenTelemetry
```

Comment:
Doesn't this need to part of the API? Presumably a user could have code such as:

```Ruby
span.add_link(-> { OpenTelemetry::Trace::Link.new(ctx) }) 
```
and this should "work" with the minimal implementation. Events will be similar.

Reply:
I suppose that would "work" so long as the proc isn't executed.

Reply:
With this API, that turns into:
```ruby
span.add_link(-> { tracer.create_link(ctx) })
```